python/redis_work_queue/workqueue.py

```diff
@@ -37,6 +37,37 @@ def add_item(self, db: Redis, item: Item) -> None:
         pipeline = db.pipeline()
         self.add_item_to_pipeline(pipeline, item)
         pipeline.execute()
+    
+    def add_new_item(self, db: Redis, item: Item) -> bool:
+        """Adds an item to the work queue only if an item with the same ID doesn't already exist.
+
+        This method uses WATCH to add the item atomically. The db client passed must not be used by
+        anything else while this method is running.
+
+        Returns a boolean indicating if the item was added or not. The item is only not added if it
+        already exists or if an error (other than a transaction error, which triggers a retry) occurs."""
+        pipeline = db.pipeline(transaction=True)
+        while True:
+            try:
+                pipeline.watch(self._main_queue_key, self._processing_key)
```

Comment:
Does this actually do the `WATCH` now, or does this only do the `WATCH` when the pipeline is executed? I don't know the answer.